lib/broccoli/amd-excluder.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Funnel = require('broccoli-funnel');
+const walkSync = require('walk-sync');
+
+class AmdExcluder extends Funnel {
+  constructor(inputNode, options) {
+    super(inputNode, {
+      exclude: [],
+      annotation: options.annotation,
+    });
+
+    this.options = options;
+  }
+
+  build() {
+    let inputPath = this.inputPaths[0];
+
+    let files = walkSync(inputPath, {
+      directories: false,
+      globs: [`**/*.js`],
+    });
```

Comment:
does `build()` have to be sync? otherwise it might make sense to read and process several files in parallel 🤔

Reply:
Doesn't have to be. Which package should I use for async walking?

Reply:
I guess `walkSync` is okay, but the loop afterwards could possibly be concurrent

Reply:
Changed to async readFile. Was this what you were thinking?